Update GUIDELINES.md allowing non-US English in search terms (#2693)
openstreetmap/id-tagging-schema: GUIDELINES.md

The user interface must be clear, concise, and easy to use, leaving no room for misunderstandings.

- Define the tags required on an object to trigger the preset.

> - Choose a name label using American English.

Author: ```suggestion
- Choose a name label using American English. This dialect is used for base version of text (translation for other languages and dialects are available).
```

Author: but it may be too wordy to be worth including

> - Add search terms - these should typically be American English, but they may rarely include terms from other English dialects.

Author: ```suggestion
- Add search terms - these should typically be American English, but they may rarely include terms from other English dialects. Terms from other dialects must not cause confusion.
```

Author: @1ec5 would it resolve your concern?

Reviewer: Yes, that’s probably the best we can do purely in this repository.

The user interface must be clear, concise, and easy to use, leaving no room for misunderstandings.

- Define the tags required on an object to trigger the preset.
- Choose a name label using American English.

> - Add search terms - these may include limited English content outside American English.

Reviewer: “Category” dropped out of this change – is it still required?

Author: Yes,  as I have no idea what category is in this context and it is not mentioned in schema.md

Unless I missed something?

We have categories but are these set in presets?

(I think it was mentioned in commit description)

Reviewer: I assumed that it was a suggestion to add the preset to a relevant [preset category](https://github.com/openstreetmap/id-tagging-schema/tree/1e7cbfd4d5d78b098bccda043d77cf240bdf224d/data/preset_categories). Not every preset should be part of a category, but it wouldn’t hurt to build out the categories more. Most new mappers don’t know they can search for something that isn’t in the initial listing.

Author: I restored mention of categories, but without its naming as it is about preset creation not about category creation

Reviewer: OK, sounds good. (I don’t have permission to review this PR as requested.)

Author: I think you should be still able to leave a comment

> - Add search terms - these may include limited English content outside American English.

Reviewer 2: ```suggestion
- Add search terms - these should typically be American English, but they may include limited terms from other English dialects.
```

?
